realm/realm-library/src/objectServer/java/io/realm/mongodb/functions/Functions.java

```diff
@@ -96,7 +97,34 @@ protected Functions(RealmUser user, CodecRegistry codecRegistry) {
      * @see RealmAppConfiguration#getDefaultCodecRegistry()
      */
     public <T> T callFunction(String name, List<?> args, Class<T> resultClass) {
-        return callFunction(name, args, resultClass, defaultCodecRegistry);
+        return callFunction(name, args, defaultCodecRegistry, defaultCodecRegistry.get(resultClass));
+    }
+
+    /**
+     * Call a MongoDB Realm function synchronously with custom result decoder.
+     * <p>
+     * The arguments will be encoded with the default codec registry encoding.
+     *
+     * @param name Name of the Stitch function to call.
+     * @param args Arguments to the Stitch function.
+     * @param resultDecoder The decoder used to decode the result.
+     * @param <T> The type that the response will be decoded as using the {@code resultDecoder}
+     * @return Result of the Stitch function.
+     *
+     * @throws ObjectServerError if the request failed in some way.
+     * @throws org.bson.codecs.configuration.CodecConfigurationException if the {@code codecRegistry}
+     * does not provide codecs for the argument or {@code resultClass}.
+     * @throws org.bson.BSONException is an error occurred during BSON processing.
+     *
+     * @see #callFunction(String, List, Class, CodecRegistry)
+     * @see RealmAppConfiguration#getDefaultCodecRegistry()
+     */
+    public <T> T callFunction(String name, List<?> args, Decoder<T> resultDecoder) {
+        return callFunction(name, args, defaultCodecRegistry, resultDecoder);
+    }
+
+    <T> T callFunction(String name, List<?> args, CodecRegistry codecRegistry, Decoder<T> resultDecoder) {
+        return invoke(name, args, codecRegistry, resultDecoder);
```

Comment:
Is there a reason you are not calling `invoke` directly from the public `callFunction` methods?

Reply:
No, did not streamline the various versions in the first go and did not catch that it ended up being a passthrough method in the end. 👍